File: commit.py
import os

# Ruta al programa ejecutable (por ejemplo, /bin/ls en sistemas Unix)
programa = "/bin/ls"

# Argumentos para el programa (el primero suele ser el nombre del programa)
args = {["git", "add", "."], ["git", "commit", "-m", sys.argv[1]], ["git", "push"]}

# Variables de entorno (puedes copiar las existentes con os.environ)
entorno = os.environ.copy()

# Ejecutar el programa
for x < 3:
	os.execve(args, args, entorno)

File: commit.cpp
#include<iostream>
#include<fcntl.h>
#include<unistd.h>
#include<sys/wait.h>

int main(int argc, char **argv, char **envp)
{
	if (argc != 2)
		return 1;
	char	s[3][4][100] = {{"git", "add", "."}, {"git", "commit", "-m", argv[1]}, {"git", "push"}};
	for (int i = 0; i < 3; i++)
	{
		pid_t pid = fork();
		if (pid == 0)
			execve(s[i][0], &s[i][0], envp);
		waitpid(pid, NULL, 0);
	}
	return 0;
}